### PR TITLE
Refacto datasets

### PR DESCRIPTION
**Pull request by arthurPignet**
Change the architecture around the dataset in a object-oriented way

### CODE REVIEW

File: tests/unit_tests.py [hunk 1]
```diff
@@ -181,21 +137,27 @@ def create_Scenario(create_Dataset, iterate_samples_split_option):
...
+    scenar.compute_batch_sizes()
+    scenar.data_corruption()
+
+    return scenar
 
 
 @pytest.fixture(scope="class")
 def create_Contributivity(create_Scenario):
     scenar = create_Scenario
     contri = Contributivity(scenario=scenar)
```
Comment (RomainGoussault) [inline tests/unit_tests.py:158]: contrib ?

File: tests/unit_tests.py [hunk 2]
```diff
@@ -181,21 +137,27 @@ def create_Scenario(create_Dataset, iterate_samples_split_option):
         **params, experiment_path=experiment_path, scenario_id=0, repeats_count=1
     )
 
-    scenar.partners_list = create_partners_list(
-        scenar.dataset.name, scenar.partners_count
-    )
-
     scenar.mpl = scenar.multi_partner_learning_approach(scenar, is_save_data=True)
```
Comment (RomainGoussault) [inline tests/unit_tests.py:140]: scenario maybe

File: tests/unit_tests.py [hunk 3]
```diff
@@ -298,71 +228,49 @@ def test_Contributivity(self, create_Contributivity):
...
+        assert len(data.x_val) == len(data.y_val), "Number of val label is not equal to the number of data"
+        assert len(data.x_test) == len(data.y_test), "Number of test label is not equal to the number of data"
+
+        if data.num_classes > 2:
+            assert data.y_train[np.random.randint(len(data.y_train))].shape == (data.num_classes,)
+            assert data.y_val[np.random.randint(len(data.y_val))].shape == (data.num_classes,)
+            assert data.y_test[np.random.randint(len(data.y_test))].shape == (data.num_classes,)
+        assert data.x_train[np.random.randint(len(data.x_train))].shape == data.input_shape
+        assert data.x_test[np.random.randint(len(data.x_test))].shape == data.input_shape
+        assert data.x_val[np.random.randint(len(data.x_val))].shape == data.input_shape
```
Comment (RomainGoussault) [inline tests/unit_tests.py:263]: remove randomness.

File: tests/unit_tests.py [hunk 4]
```diff
@@ -298,71 +228,49 @@ def test_Contributivity(self, create_Contributivity):
...
+
+    def test_generate_new_model(self, create_all_datasets):
+        dataset = create_all_datasets
+        model1 = dataset.generate_new_model()
```
Comment (RomainGoussault) [inline tests/unit_tests.py:267]: rename to model

**Review by RomainGoussault — APPROVED** [review]
Looks good !

**Review by Thomas-Galtier — APPROVED** [review]

**Pull request merged**